src/network/networkd-address.c

```diff
@@ -482,10 +482,10 @@ int link_has_ipv6_address(Link *link, const struct in6_addr *address) {
         a->family = AF_INET6;
         a->in_addr.in6 = *address;
 
-        return address_get(link, a, NULL) >= 0;
+        return address_get(link, a, ret);
 }
 
-static int link_get_ipv4_address(Set *addresses, const struct in_addr *address, Address **ret) {
+static int set_get_ipv4_address(Set *addresses, const struct in_addr *address, Address **ret) {
```

Comment:
Hmm, the name `set_get_…` is rather confusing. I had to read the code to understand that `set` is a noun, not a verb here. Maybe call this `addresses_get_ipv4_address` ?